helper/resource/testing_new_import_state.go

```diff
@@ -228,16 +229,40 @@ func testStepNewImportState(ctx context.Context, t testing.T, helper *plugintest
 		if err := runPlanChecks(ctx, t, plan, step.ImportPlanChecks.PreApply); err != nil {
 			return err
 		}
-	} else {
-		err = runProviderCommand(ctx, t, func() error {
-			return importWd.Import(ctx, resourceName, importId)
-		}, importWd, providers)
-		if err != nil {
-			return err
+
+		{
+			if kind.resourceIdentity() {
+				err := runProviderCommandApply(ctx, t, wd, providers)
+				if err != nil {
+					return fmt.Errorf("applying plan with import config: %s", err)
+				}
+
+				newStateJSON, err := runProviderCommandGetStateJSON(ctx, t, wd, providers)
+				if err != nil {
+					return fmt.Errorf("getting state after applying plan with import config: %s", err)
+				}
```

Comment:
I was testing this out and I believe these commands should be run against `importWd`?

Reply:
Yes, 💯. I've pushed a new commit to correct this.